src/zh/roumanwu/src/eu/kanade/tachiyomi/extension/zh/roumanwu/Roumanwu.kt

```diff
@@ -36,41 +42,100 @@ class Roumanwu : HttpSource(), ConfigurableSource {
     private val json: Json by injectLazy()
 
     override fun popularMangaRequest(page: Int) = GET("$baseUrl/home", headers)
-    override fun popularMangaParse(response: Response) = response.nextjsData<HomePage>().getPopular().toMangasPage()
+    override fun popularMangaNextPageSelector(): String? = null
+    override fun popularMangaSelector(): String = "main > div.px-0 > div > div:not(.p-3):not(:last-child):not(:nth-child(4)) > div.grid > div > a"
```

Comment:
these kind of selectors can be broken easily... (applies to almost all of the selectors here).
Ik these are tailwind selectors and aren't cleanly scrapable, but consider using pseudo selectors to make it a bit cleaner.
try `.grid a[href*=/books/]` 

also take a look at the [documentation](https://jsoup.org/apidocs/org/jsoup/select/Selector.html) to learn about pseudo selectors

Reply:
<img width="281" alt="image" src="https://github.com/user-attachments/assets/3f9ff745-ff3c-4b94-9a87-b72d779718c0">

Because under the `div.px-1`, the 2nd, 3rd, 5th are `hotest`, `daily top`, `weekly top`. These three `grid` is for popular.
And the 4th one is `Latest`.
They have no difference in css.
The best i can try is `div.px-1 > div:not(:first-child):not(:last-child):not(:nth-child(4)) .grid a[href*=/books/]` for popular and `div.px-1 > div:nth-child(4) .grid a[href*=/books/]` for latest.
Or if we do `.grid a[href*=/books/]` and set supportsLatest to false, since everything is inside the popular.

Reply:
or we do `div:contains(最近更新) .grid a[href*=/books/]`?

Reply:
`div:contains(正熱門), div:contains(今日最佳), div:contains(本週熱門) .grid a[href*=/books/]`
`div:contains(最近更新) .grid a[href*=/books/]`
seems to work

Reply:
```
div.px-1 > div:contains(正熱門) .grid a[href*=/books/],
div.px-1 > div:contains(今日最佳) .grid a[href*=/books/],
div.px-1 > div:contains(本週熱門) .grid a[href*=/books/]
```

`div.px-1 > div:contains(最近更新) .grid a[href*=/books/]`

Reply:
`div:matches(最近更新|本週熱門|今日最佳|正熱門) .grid a[href*=/books/]`